{{ cookiecutter.__project_name_kebab }}/.github/workflows/publish.yml

```diff
@@ -32,10 +31,23 @@ jobs:
       - name: Build
         run: python -m flit build
 
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./dist
+
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read # to fetch code (actions/checkout)
```

Comment:
I wonder if this is needed since we're not using actions/checkout.

Other than this :shipit:

Reply:
Just to be clear: Removing that line doesn't remove the permission. `contents: read` just re-states the default.

We've discussed it before in

https://github.com/wagtail/cookiecutter-wagtail-package/pull/29#discussion_r1038042000
and
https://github.com/wagtail/cookiecutter-wagtail-package/pull/28#discussion_r1038057635

but

```yaml
permissions:
  contents: read  # default
  metadata: read  # default

jobs:
  build:
    runs-on: ubuntu-latest
    ...
```

and

```yaml
# nothing

jobs:
  build:
    runs-on: ubuntu-latest
    ...
```

are functionally the same.

That said, I gave it a try and the publish workflow will actually run with `contents: none` (which _does_ revoke the read permission) - it is not needed to download the artifact - so we might as well give it one less permission.

Merging with that change